### PR TITLE
Initial add of a contributor tool directory for various scripts

### PR DESCRIPTION
**Pull request by ekluzek**
### Description of changes

Initial add of a contributor tool directory for various scripts to be used fo pre and post processing, as well as case management of CTSM.

### Specific notes

Contributors other than yourself, if any: @swensosc @olyson @wwieder 

CTSM Issues Fixed (include github issue #): #619

Are answers expected to change (and if so in what way)? No

Any User Interface Changes (namelist or namelist defaults changes)? No

Testing performed, if any: None

### CODE REVIEW

**Comment by olyson** [comment]
There is an updated version of the SpinupStability script here:

/gpfs/u/home/oleson/lnd_diag/run/SpinupStability_v8.ncl

An updated version of the run_historical script here:

/gpfs/u/home/oleson/run_hist_1850_files/BGC/run_clm_historical.v7.csh

and an updated version of the run_clmtowers script here:

/gpfs/fs1/work/oleson/release-clm5.0.12/tools/PTCLM/run_clmtowers.csh

**Comment by ekluzek** [comment]
Some of the scripts have usernames hardwired in, and personal paths as well. The tower script points to data that @olyson has so in cases like that we'll need to leave that in. But, in other cases it should create data and expect data according to the user's name rather than hardcoded to a specific individual.

Also we want it setup to run the Tutorial cases out of the box, without having to make changes to it. We still want to educate users on how you make those changes, but don't want them to HAVE to make changes to get something to work.

They might also be hardwired to only work on cheyenne. That's probably OK for now. But, that could be an issue for general use.

File: tools/contrib/singlept [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+#  Import libraries
+import sys
+import os
+from getpass import getuser
+import string
+import subprocess
+import numpy as np
+import xarray as xr
+
+def mprint(mstr):
```
Comment (billsacks) [inline tools/contrib/singlept]: Instead of this, use

```
from __future__ import print_function
```

Then just use

```
print(...)
```

**Comment by ekluzek** [comment]
OK, I made some changes to get it working for me. And added some more printing about what it's doing. I'm going to approve this as it is for now. But, we will likely continue working on it.

**Pull request merged**